patches/0005-LPS-112982-Add-additional-resource-URL-parameters.patch

```diff
@@ -0,0 +1,73 @@
+From 28a9f881549502741051f59be88e0e74a027a3c8 Mon Sep 17 00:00:00 2001
+From: Julien Castelain <julien.castelain@liferay.com>
+Date: Mon, 6 Jul 2020 12:57:00 +0200
+Subject: [PATCH 5/5] LPS-112982 Add additional resource URL parameters
+
+---
+ core/ckeditor_base.js | 27 ++++++++++++++++++++-------
+ core/config.js        |  9 +++++++++
+ 2 files changed, 29 insertions(+), 7 deletions(-)
+
+diff --git a/core/ckeditor_base.js b/core/ckeditor_base.js
+index 3fb18f5f7..32d1dadd5 100644
+--- a/core/ckeditor_base.js
++++ b/core/ckeditor_base.js
+@@ -171,15 +171,28 @@ if ( !window.CKEDITOR ) {
+ 			 * @returns {String} The full URL.
+ 			 */
+ 			getUrl: function( resource ) {
+-				// If this is not a full or absolute path.
+-				if ( resource.indexOf( ':/' ) == -1 && resource.indexOf( '/' ) !== 0 )
+-					resource = this.basePath + resource;
++				var resourceUrl = resource.startsWith(this.basePath)
++						? new URL(resource)
++						: new URL(this.basePath + resource);
+ 
+-				// Add the timestamp, except for directories.
+-				if ( this.timestamp && resource.charAt( resource.length - 1 ) != '/' && !( /[&?]t=/ ).test( resource ) )
+-					resource += ( resource.indexOf( '?' ) >= 0 ? '&' : '?' ) + 't=' + this.timestamp;
++				if (!resourceUrl.toString().endsWith('/')) {
++					if (this.timestamp && !resourceUrl.searchParams.has('t')) {
++						resourceUrl.searchParams.set('t', this.timestamp);
++					}
++
++					if (CKEDITOR.ADDITIONAL_RESOURCE_PARAMS && typeof CKEDITOR.ADDITIONAL_RESOURCE_PARAMS === 'object') {
```

Comment:
`typeof null` is `'object'`, and `Object.entries(null)` will throw a `TypeError`.

As a rule of thumb, if you *ever* see `typeof blah === 'object'` in a codebase, it is probably wrong, unless accompanied by a guard (eg. `blah && typeof blah === 'object'`).

Reply:
Sigh. If this comment doesn't appear to make any sense, it's because I didn't see that you already had `CKEDITOR.ADDITIONAL_RESOURCE_PARAMS &&` — I blame GitHub's very narrow UI.

Reply:
yes I remember we actually talked about this a few time. 
You can blame GitHub's UI ... or me for doing force push all the time